merge unnecessary Docker layers

e-koma opened the pull request:
- There are many unnecessary `ENV` / `RUN` layers, so merged.

## Result
- delete 30MB image size
  - [vvakame/review:3.0](https://hub.docker.com/r/vvakame/review/tags)  806MB
  - [ekoma/docker-review](https://cloud.docker.com/repository/docker/ekoma/docker-review/tags) 776MB

Commit:
delete unnecessary space

Commit:
merge ENV layers

Commit:
sort the packages alphabetically

Commit:
merge RUN layers

vvakame commented:
ありがとうございます。
ENVやらRUNやらを圧縮して容量を削減したい、という方針はOKです。
ただ、諸々の記述順序は変えたくないところです。
この行はざっくりtex入れてて…次の行でreviewが必要なもの入れて…グラフ系ツール入れて… みたいな感じになっています。これを崩すとなんでこれ入れてるんだっけ…？となってメンテできなくなるからです。
また、update.shもメンテし、各バージョン毎のDockerfileにも変更を反映できるようにしていただけると嬉しいです。

e-koma commented:
レビューありがとうございます。
- 記述順序を元通りに合わせました。
- update.shを実行して各versionのDockerfileも反映しました。
- 修正後の状態でdocker buildしてビルドが成功すること確認しました。

（PR出しておいてこんなコメント言うのも恐縮なのですが
　修正後の状態がメンテしづらいと思われるようでしたら、
　ガッツリRUNコマンドを削るのではなく、
　ENV等の本当に最低限のLayerだけ削るだけにしておく、のでもありかなと思いました）

[追記]
最後にforce-pushの痕跡を残してしまいましたが、コメント以降にpushしてしまったcommitを直したかったためです。

Commit:
move install commands order

Commit:
disable Emplty continuation lines Warning

because the follow warning is apper.
`[WARNING]: Empty continuation lines will become errors in a future release.`

Commit:
apply Dockerfile to each versions with update.sh

vvakame commented:
ENV自体のsyntaxを変えてしまっているので、update.shの記述を修正しないといけなさそうですね。
例えば、 Re:VIEW 2.4用のディレクトリでも `REVIEW_VERSION=3.0.0` のままになっています。

たぶん何らかのツールを使っておられるのかと思うのですが、ちょっと僕の好みにはあわなさそうです。
しかしイメージサイズを削減したいのはわかるなぁ…という気持ちです。

e-koma commented:
あぁぁそういう意味でしたか。。失礼しました。

e-koma commented:
今回、Re:VIEWを使わせていただくにあたって環境構築する際に、
pullするLayerが多くて気になったので、手元直したら容量削減できたのでPRを送った、という経緯です。
ただ、そもそも容量削減効果がめっちゃ大きいわけではないので、

> ちょっと僕の好みにはあわなさそうです

というのであれば、本PR、Closeしちゃってもいいかなという気持ちになりました。
（いったんENVだけ元通りに戻します）

Commit:
rollback ENV layers

vvakame commented:
うーん、ではすみませんが一旦閉じてしまいます…！
後ほど振り返ってやるかもしれません。
